xclim/core/utils.py

```diff
@@ -162,3 +164,208 @@ def sfcwind_2_uas_vas(wind: xr.DataArray = None, windfromdir: xr.DataArray = Non
+                    + 1.673952e-5 * tas ** 2
+                    + 2.433502 * np.log(tas)  # numpy's log is ln
+                ),
+                100
+                * np.exp(  # Where ref_is_water is False (thus ref is ice)
+                    -6024.5282 / tas
+                    + 24.7219
+                    + 1.0613868e-2 * tas
+                    + -1.3198825e-5 * tas ** 2
+                    + -0.49382577 * np.log(tas)
+                ),
+            )
+            source = "Sonntag (1990)"
+        elif method in ["tetens30", "TE30"]:
+            e_sat = xr.where(
+                ref_is_water,
+                610.78 * np.exp(17.269388 * (tas - 273.16) / (tas - 35.86)),
+                610.78 * np.exp(21.8745584 * (tas - 273.16) / (tas - 7.66)),
+            )
+            source = "Tetens (1930)"
+        elif method in ["goffgratch46", "GG46"]:
+            Tb = 373.16  # Water boiling temp [K]
+            eb = 101325  # e_sat at Tb [Pa]
+            Tp = 273.16  # Triple-point temperature [K]
+            ep = 611.73  # e_sat at Tp [Pa]
+            e_sat = xr.where(
+                ref_is_water,
+                eb
+                * 10
+                ** (
+                    -7.90298 * ((Tb / tas) - 1)
+                    + 5.02808 * np.log10(Tb / tas)
+                    + -1.3817e-7 * (10 ** (11.344 * (1 - tas / Tb)) - 1)
+                    + 8.1328e-3 * (10 ** (-3.49149 * ((Tb / tas) - 1)) - 1)
+                ),
+                ep
+                * 10
+                ** (
+                    -9.09718 * ((Tp / tas) - 1)
+                    + -3.56654 * np.log10(Tp / tas)
+                    + 0.876793 * (1 - tas / Tp)
+                ),
+            )
+            source = "Goff and Gratch (1946)"
+        elif method in ["wmo08", "WMO08"]:
+            e_sat = xr.where(
+                ref_is_water,
+                611.2 * np.exp(17.62 * (tas - 273.16) / (tas - 30.04)),
+                611.2 * np.exp(22.46 * (tas - 273.16) / (tas - 0.54)),
+            )
+            source = "WMO (2008)"
+        else:
+            raise NotImplementedError(f"Method {method} is not (yet) implemented.")
+
+        w = huss / (1 - huss)
+        w_sat = 0.62198 * e_sat / (ps - e_sat)
+        rh = 100 * w / w_sat
+
+        desc = f"Relative humidity computed from temperature, specific humidity and pressure. The saturation vapor pressure was calculated according to {source}."
+        if ice_thresh is not None:
+            desc += f" The computation was done in reference to ice for temperatures below {ice_thresh}."
+
+    from xclim import __version__
+
+    if invalid_values == "clip":
+        rh = rh.clip(0, 100)
+    elif invalid_values == "fill":
+        rh = rh.where((rh <= 100) & (rh >= 0))
+
+    rh.name = "rh"
+    rh.attrs["standard_name"] = "relative_humidity"
+    rh.attrs["long_name"] = "Relative Humidity"
+    rh.attrs["description"] = desc
+    rh.attrs["units"] = "%"
+    rh.attrs[
```

Comment:
History handling should probably not be done by individual indices, it should be a generic operation in the Indicator class. If there is custom information to pass from the indice, I think we want to clarify how it should be done in general. Rule so far was that indices did not mess with attributes, except for units.